sound/soc/sof/ipc4.c

```diff
@@ -631,11 +631,15 @@ static void sof_ipc4_rx_msg(struct snd_sof_dev *sdev)
 			return;
 
 		ipc4_msg->data_size = data_size;
-		snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		err = snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		if (err < 0) {
+			dev_err(sdev->dev, "failed to read IPC notification data: %d\n", err);
+			goto out;
+		}
 	}
 
 	sof_ipc4_log_header(sdev->dev, "ipc rx done ", ipc4_msg, true);
-
+out:
```

Comment:
Does this applies at all?
We are now dumping the payload if data_size is not NULL and we might dump invalid data if the snd_sof_ipc_msg_data() failed.

Reply:
> Does this applies at all? We are now dumping the payload if data_size is not NULL and we might dump invalid data if the snd_sof_ipc_msg_data() failed.

Yes, it didn't conflict. But, yes, it might dump invalid data if the snd_sof_ipc_msg_data() failed. So, return err immediately.